script.js

```diff
@@ -0,0 +1,366 @@
+const API_URL = 'https://raw.githubusercontent.com/pahanan/pahan_an.github.io/main/api/data.json';
+const INGREDIENTS_API_URL = 'https://raw.githubusercontent.com/pahanan/pahan_an.github.io/main/api/ingredient-prices.json';
+let recipes = [];
+let ingredientPrices = {};
+
+async function fetchRecipes() {
+  try {
+    const response = await fetch(API_URL);
+
+    if (!response.ok) {
+      throw new Error(`Error!${response.status}`);
+    }
+
+    recipes = await response.json();
+    console.log("Error!", recipes);
+    await fetchIngredientPrices(); 
+    enrichRecipesWithPrices(); 
+    renderRecipes(recipes);
+  } catch (error) {
+    console.error('Error!', error);
+  }
+}
+
+async function fetchIngredientPrices() {
+  try {
+    const response = await fetch(INGREDIENTS_API_URL);
+
+    if (!response.ok) {
+      throw new Error(`Error!  ${response.status}`);
+    }
+
+    ingredientPrices = await response.json();
+  } catch (error) {
+  }
+}
+
+function enrichRecipesWithPrices() {
+  recipes.forEach((recipe) => {
+    recipe.ingredients.forEach((ingredient) => {
+      const price = ingredientPrices[ingredient.name] || 'N/A';
+      ingredient.price = price;
+    });
+  });
+}
+
+function renderRecipes(recipes) {
```

Comment:
I like that you called this `renderRecipes`, indicating that this is the function providing the bulk of the dynamic part of the page.